app/models/manageiq/providers/ansible_tower/shared/automation_manager/configuration_script.rb

```diff
@@ -1,87 +1,13 @@
 module ManageIQ::Providers::AnsibleTower::Shared::AutomationManager::ConfigurationScript
   extend ActiveSupport::Concern
 
-  module ClassMethods
-    NAME_MAP = {
-      'create' => 'Creating',
-      'update' => 'Updating',
-      'delete' => 'Deleting'
-    }.freeze
-    private_constant :NAME_MAP
-
-    def create_in_provider(manager_id, params)
-      manager = ExtManagementSystem.find(manager_id)
-      job_template = manager.with_provider_connection do |connection|
-        connection.api.job_templates.create!(params)
-      end
-
-      # Get the record in our database
-      # TODO: This needs to be targeted refresh so it doesn't take too long
-      task_ids = EmsRefresh.queue_refresh_task(manager)
-      task_ids.each { |tid| MiqTask.wait_for_taskid(tid) }
-
-      find_by!(:manager_id => manager.id, :manager_ref => job_template.id)
-    rescue AnsibleTowerClient::Error => err
-      begin
-        raise err.class, JSON.parse(err.message).values.flatten.join(", ")
-      rescue JSON::ParserError
-        raise err
-      end
-    end
+  include ProviderObjectMixin
 
-    def update_in_provider(manager_id, params)
-      manager = ExtManagementSystem.find(manager_id)
-      manager.with_provider_connection do |connection|
-        connection.api.job_templates.find(params.delete(:manager_ref)).update_attributes!(params)
-      end
-
-      # Get the record in our database
-      # TODO: This needs to be targeted refresh so it doesn't take too long
-      EmsRefresh.queue_refresh(manager)
-    end
-
-    def delete_in_provider(manager_id, params)
-      manager = ExtManagementSystem.find(manager_id)
+  module ClassMethods
+    def provider_collection(manager)
       manager.with_provider_connection do |connection|
-        connection.api.job_templates.find(params[:manager_ref]).destroy!
+        connection.api.job_templates
       end
-
-      # Get the record in our database
-      # TODO: This needs to be targeted refresh so it doesn't take too long
-      EmsRefresh.queue_refresh(manager)
-    end
-
-    def delete_in_provider_queue(manager_id, params, auth_user = nil)
-      operate_in_provider_queue('delete', manager_id, params, auth_user)
-    end
-
-    def update_in_provider_queue(manager_id, params, auth_user = nil)
-      operate_in_provider_queue('update', manager_id, params, auth_user)
-    end
-
-    def create_in_provider_queue(manager_id, params, auth_user = nil)
```

Comment:
Thanks Drew. I've now added the optional param `auth_user` to the 3 `_queue` methods.